packages/gatsby-source-wordpress/src/steps/create-schema-customization/build-types.js

```diff
@@ -144,7 +131,11 @@ const objectType = typeBuilderApi => {
       .filter(interfaceType => {
         const interfaceTypeSettings = getTypeSettingsByType(interfaceType)
 
-        return !interfaceTypeSettings.exclude && fieldOfTypeWasFetched(type)
+        return (
+          !interfaceTypeSettings.exclude &&
+          fieldOfTypeWasFetched(type) &&
+          fieldOfTypeWasFetched(interfaceType)
```

Comment:
Interface types that were not fetched (added during query building) shouldn't appear in the schema, so this filters them out